### PR TITLE
Move Lint/Eval to Security/Eval

### PR DESCRIPTION
**Pull request by cyberdelia**
This is a proposal to move the Eval cop to Security/Eval, this is in order to have all the [security checks](https://ruby-doc.org/core-2.3.3/doc/security_rdoc.html)  under the same umbrella.

I'm not sure about the transition, I'm open to suggestions.

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] All tests are passing.
* [x] The new code doesn't generate RuboCop offenses.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

### CODE REVIEW

**Comment by Drenmi** [comment]
Makes sense, but would probably require a `CHANGELOG` entry, because it will break configurations for users referencing it in their `.rubocop.yml`. 😀

**Comment by cyberdelia** [comment]
I intentionally did not do some of the work, waiting to see if that idea gain consensus.

**Comment by bbatsov** [comment]
I'm fine with the proposed change. I actually planned to do this myself a
while back. :-)

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 ### New features
 
+* [#3820](https://github.com/bbatsov/rubocop/pull/3820): Rename `Lint/Eval` to `Security/Eval`. ([@cyberdelia][])
```
Comment (bbatsov) [inline CHANGELOG.md]: This should actually be under "Changes".

**Comment by bbatsov** [comment]
You'll also have to rebase this due to the fact your other PR just got merged.

**Comment by cyberdelia** [comment]
Rebased and changed.

**Pull request merged**